Added conditional display of resources based on device capability. Added new resources: Qtree, Filesystem and Shares for NAS devices.

anvithks opened the pull request:
**What type of PR is this?**
/kind new feature

**What this PR does / why we need it**:
This PR adds support for conditionally displaying the resources supported by a device. SODA Delfin supports SAN, NAS and  devices with Unified capabilities. Some devices may not have all the resources supported and the Dashboard should not display the resources for such devices.
Currently Delfin supports: Storage Pools, Volumes, Controllers, Ports and Disks display. These are displayed for all devices in the tree view in the storage dashboard and tab view in storage details page.
With this PR we display the resource type only if the device model supports the capability. This capability is maintained as a map at the dashboard in the following format:
```
 resources:{
            volumes : ['vmax', 'unity', 'vnx', 'oceanstor', '3par', 'vsp', 'storwize_svc', 'fake_driver'],
            pools : ['vmax', 'unity', 'vnx', 'oceanstor', '3par', 'vsp', 'storwize_svc', 'fake_driver'],
            controllers : ['vmax', 'unity', 'vnx', 'oceanstor', '3par', 'vsp', 'storwize_svc', 'fake_driver'],
            ports : ['vmax', 'unity', 'vnx', 'oceanstor', '3par', 'vsp', 'storwize_svc'],
            disks : ['vmax', 'unity', 'vnx', 'oceanstor', '3par', 'vsp', 'storwize_svc', 'fake_driver'],
            qtrees : ['vmax', 'unity', 'vnx', 'oceanstor', '3par', 'vsp', 'storwize_svc', 'fake_driver'],
            filesystems : ['vmax', 'unity', 'vnx', 'oceanstor', '3par', 'vsp', 'storwize_svc', 'fake_driver'],
            shares: ['vmax', 'unity', 'vnx', 'oceanstor', '3par', 'vsp', 'storwize_svc']
        },
```
In the above Ports and Shares resources are not supported in the fake_driver devices.

The map above has to be updated with the actual support matrix based on the device capability.

This PR also adds new resource types Qtree, Filesystem and Shares to devices that support NAS capabilities.

**Which issue(s) this PR fixes**:
Fixes #530 

**Test Report Added?**:
/kind TESTED

**Test Report**:
![nas-8](https://user-images.githubusercontent.com/19162717/110826511-25d2cf80-82bb-11eb-8a4c-c7f5450e0961.png)
![nas-9](https://user-images.githubusercontent.com/19162717/110826503-24a1a280-82bb-11eb-9900-c93222ba69b8.png)
![nas-10](https://user-images.githubusercontent.com/19162717/110826494-22d7df00-82bb-11eb-87b0-72dd2dc1206d.png)
![nas-1](https://user-images.githubusercontent.com/19162717/110287502-a4561580-800c-11eb-9303-98b71e80fc7a.png)
![nas-2](https://user-images.githubusercontent.com/19162717/110287501-a3bd7f00-800c-11eb-8e33-ea4fc3a1d922.png)
![nas-3](https://user-images.githubusercontent.com/19162717/110287497-a324e880-800c-11eb-95cc-652158f206c3.png)
![nas-4](https://user-images.githubusercontent.com/19162717/110287496-a324e880-800c-11eb-9c22-d7a14ea1ebab.png)
![nas-5](https://user-images.githubusercontent.com/19162717/110287494-a28c5200-800c-11eb-897e-57548ef83d4d.png)
![nas-6](https://user-images.githubusercontent.com/19162717/110287492-a1f3bb80-800c-11eb-9372-d1c68b1f6243.png)
![nas-7](https://user-images.githubusercontent.com/19162717/110287489-a029f800-800c-11eb-9666-5f7908d02075.png)




**Special notes for your reviewer**:
The device capability matrix pushed in this PR enables all resource types for all device models.
The actual support matrix must be updated before the PR is merged or in a subsequent PR.

joseph-v commented:
Can we reorder the resource tabs to [Storage pool > Volume > Controller > Port ? Filesystem > Qtree > Share]

anvithks commented:
> Can we reorder the resource tabs to [Storage pool > Volume > Controller > Port ? Filesystem > Qtree > Share]

@joseph-v This is done. Please review the PR.

anvithks commented:
@NajmudheenCT @joseph-v @kumarashit 
I have updated this PR and added screenshots.
The resources are now conditionally displayed. 
The resources supported are maintained in the constants file in an object array as mentioned above in the PR description.
We have to update this with actual values. Please help me with the real values to update the PR.

NajmudheenCT commented on src/app/shared/utils/consts.ts:
```diff
@@ -171,6 +174,16 @@ export const Consts = {
             }
         ],
+        resources:{
+            volumes : ['vmax', 'unity', 'vnx', 'oceanstor', '3par', 'vsp', 'storwize_svc', 'fake_driver'],
```
resource list need to be based on the current backend support.

current backend support is 
resources:{
            volumes : ['vmax', 'unity', 'vnx', 'oceanstor', '3par', 'vsp', 'storwize_svc', 'fake_driver'],
            pools : ['vmax', 'unity', 'vnx', 'oceanstor', '3par', 'vsp', 'storwize_svc', 'fake_driver'],
            controllers : ['oceanstor', 'fake_driver']
            ports : ['oceanstor', 'fake_driver']
            disks : ['oceanstor', 'fake_driver'],
            qtrees : ['oceanstor', 'fake_driver'],
            filesystems : ['oceanstor', 'fake_driver'],
            shares: ['oceanstor', 'fake_driver']
        },

anvithks replied:
Thanks @NajmudheenCT I have updated the PR with this. 
I removed `fake_driver` as it is only needed for development.

NajmudheenCT commented on src/app/shared/utils/consts.ts:
```diff
@@ -171,6 +174,16 @@ export const Consts = {
+            disks : ['vmax', 'unity', 'vnx', 'oceanstor', '3par', 'vsp', 'storwize_svc', 'fake_driver'],
+            qtrees : ['vmax', 'unity', 'vnx', 'oceanstor', '3par', 'vsp', 'storwize_svc', 'fake_driver'],
+            filesystems : ['vmax', 'unity', 'vnx', 'oceanstor', '3par', 'vsp', 'storwize_svc', 'fake_driver'],
+            shares: ['vmax', 'unity', 'vnx', 'oceanstor', '3par', 'vsp', 'storwize_svc']
```
Quotas also we need to add.. for displaying qouata we can either put it as different tab, or is it possible to associate with  filestem and qtree objects?

anvithks replied:
I do not have the details for adding Quotas. We can probably take it up as a subsequent task.

anvithks replied:
@NajmudheenCT This has been logged as an issue #539

NajmudheenCT reviewed (approved):
LGTM

joseph-v reviewed (approved):
LGTM

kumarashit reviewed (approved):
LGTM
